fastlane/fastlane
# [snapshot] Enable greater customization of the iOS status bar in snapshot screenshots

### Checklist


### Motivation and Context
The Snapshots tool allows us to use clean consistent status bars in our iOS screenshots. However, it doesn't allow us to change the network call sign from "Carrier", or use any of the multitude of options `simctl` makes available to us.

Resolves #17090


### Description
This pull request maintains the option to use a simple clean status bar as before, but also allows for complete customization of the status bar using current, or even future options made available via `simctl`.

Previously, you could add the following line to your `Snapfile` configuration file to enable a clean status bar:

```ruby
override_status_bar(true)
```

With this pull request, you can add a `override_status_bar_arguments` line with arguments to be passed to the `xcrun simctl` command used to modify the status bar. This give you full control to customize the status bar appropriately for your app. Example:

```ruby
override_status_bar(true)
override_status_bar_arguments("--time 9:41 --dataNetwork lte --cellularMode active --cellularBars 4 --batteryState charging --batteryLevel 75 --operatorName TELUS")
```


### Testing Steps

Create a `Snapfile` for use with the `Snapshot` tool. Include the following lines:

```ruby
override_status_bar(true)
#override_status_bar_arguments("") # Produces the same clean status bar that would have been produced before this PR 
#override_status_bar_arguments("--time 9:41") # Sets the time to 9:41, leaves all other options unchanged
#override_status_bar_arguments("--time 9:41 --dataNetwork lte --cellularMode active --cellularBars 4 --batteryState charging --batteryLevel 75 --operatorName TELUS") # Set the time to 9:41, battery changing at 75%, on the TELUS LTE network
```

Uncomment one of the `override_status_bar_arguments` lines at a time and compare the results on the status bar to the expected results.

Also try commenting out the `override_status_bar` line, or set it to `false` to ensure the `override_status_bar_arguments` is ignored in those cases.
